packages/@aws-cdk/aws-ecs/test/base/base-service.test.ts

```diff
@@ -0,0 +1,133 @@
+import * as cdk from '@aws-cdk/core';
+import * as cloudmap from '@aws-cdk/aws-servicediscovery';
+import * as ecs from '../../lib';
+
+describe('When associating with service registry', () => {
```

Comment:
Given the suggestion of keeping `determineContainerNameAndPort` private, it would probably make more sense to have unit tests based on the developer experience using the non-abstract classes rather than unit testing a private method. For example, I like the test you added in `ec2-service.test.ts` -- perhaps we can do something similar for error cases there?

Reply:
Sounds good to me. I'll get on this tomorrow when I'm back at work.